course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -0,0 +1,237 @@
+            $httpBackend.flush(2);
+        });
+
+        it('should set the course instance id', function () {
+            expect(dc.courseInstanceId).toEqual($routeParams.courseInstanceId);
+        });
+
+    });
+
+    describe('setCourseInstance', function () {
+        var ci;
+        var members;
+        beforeEach(function () {
+            // if we want to pull the instance id from $routeParams, this has
+            // to be in a beforeEach(), can't be in a describe().
+            ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Title',
+                course: {
+                    school_id: 'abc',
+                    registrar_code_display: '2222'
+                }
+            };
+            members = {
+                count: 100
+            };
+        });
+
+        it('should work when CourseInstance is empty', function () {
+            var dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
+            $httpBackend.flush(2);
+            expect(dc.courseInstance['title']).toEqual(ci.title);
+            expect(dc.courseInstance['school']).toEqual(ci.course.school_id.toUpperCase());
+            expect(dc.courseInstance['members']).toEqual(100);
+        });
+
+        it('should work when CourseInstance returns invalid id', function () {
+            controller = $controller('DetailsController', {$scope: scope});
+            ci.course_instance_id = 12345;
+            spyOn($log, 'error');
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
+            $httpBackend.flush(2);
+            expect($log.error).toHaveBeenCalled();
+        });
+    });
+
+    describe('stripQuotes', function () {
+        var ci;
+        var members;
+        beforeEach(function () {
+            // if we want to pull the instance id from $routeParams, this has
+            // to be in a beforeEach(), can't be in a describe().
+            ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: '"Test Title"',
+                course: {
+                    school_id: 'abc',
+                    registrar_code_display: '2222'
+                }
+            };
+            members = {
+                count: 100
+            };
+        });
+
+        it('should strip the quotes from the string', function () {
+            controller = $controller('DetailsController', {$scope: scope});
+            result = dc.stripQuotes(ci.title);
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
```

Comment:
The `$httpBackend` cleanup has nothing to do with what we're testing...can we move it into an `afterEach` so it's more obvious what's actually involved in the test?